ssz/sedes/list.py

```diff
@@ -0,0 +1,119 @@
+from collections.abc import (
+    Iterable,
+)
+
+from ssz.exceptions import (
+    DeserializationError,
+    SerializationError,
+)
+from ssz.sedes import (
+    address,
+    boolean,
+    hash32,
+    uint32,
+)
+
+
+class List:
+    """
+    A sedes for lists.
+
+    WARNING: Avoid sets if possible, may not always lead to expected results
+    (This is because iteration in sets doesn't always happen in the same order)
+    """
+    LENGTH_BYTES = 4
+
+    def __init__(self, element_sedes=None, empty=False):
+        if element_sedes and empty:
+            raise ValueError(
+                "Either one of Element Sedes or Empty has to be specified"
+            )
+
+        elif not element_sedes and not empty:
+            raise ValueError(
+                "Either Element Sedes or Empty has to be specified"
+            )
+
+        # This sedes object corresponds to each element of the iterable
+        self.element_sedes = element_sedes
+        # This empty bool indicates whether this sedes is meant for empty lists
+        self.empty = empty
+
+    def serialize(self, val):
+        if (
+                not isinstance(val, Iterable) or
+                isinstance(val, bytes) or
+                isinstance(val, bytes) or
```

Comment:
This check is repeated twice.

Reply:
@pipermerriam this was noticed and has been fixed as part of this PR https://github.com/ethereum/py-ssz/pull/13